File: 167. Two Sum II - Input Array Is Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        ans  = []
        low = 0
        high = len(numbers) - 1
        
        while high > low:
            if target == numbers[low] + numbers[high]:
                ans.append(low+1)
                ans.append(high+1)
                return ans
            
            elif numbers[low] + numbers[high] < target:
                low+=1
                
                
            else:
                high-=1
                
        ans.append(-1)
        ans.append(-1)
        return ans
                
                
        
                
                
            
                    
                
        

File: cpp_solutions/167. Two Sum II - Input Array Is Sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        
        vector<int> ans;
        int low = 0;
        int high = numbers.size() - 1;
        
        while (low < high){
            
            if (target == numbers[low] + numbers[high]){
                ans.push_back(low + 1);
                ans.push_back(high+1);
                return ans;
            }
            else if(numbers[low] + numbers[high] < target){
                low+=1;
            }
            else{
                high-=1;
            }
        }
        
        ans.push_back(-1);
        ans.push_back(-1);
        
        return ans;
        
    }
};
